phoenix-core/src/main/java/org/apache/phoenix/compile/DeleteCompiler.java

```diff
@@ -748,6 +756,12 @@ public Long getEstimateInfoTimestamp() throws SQLException {
         }
     }
 
+    /**
+     * Implementation of MutationPlan that is selected if
+     * 1) auto commit is enabled as well as server side delete mutations are enabled,
+     * 2) the table is not transactional,
+     * 3) there is neither WHERE clause nor LIMIT clause in the DELETE statement.
```

Comment:
WHERE clause is allowed @jinggou

Reply:
No immutable index (or client side index)

Reply:
I compare the tests for ClientSelectDeleteMutationPlan and ServerSelectDeleteMutationPlan and the difference is the WHERE clause. In what case does WHERE clause be allowed ?

Reply:
@jinggou This test uses server delete plan and it has where clause. https://github.com/apache/phoenix/blob/master/phoenix-core/src/it/java/org/apache/phoenix/end2end/DeleteIT.java#L113

You can change the test by asserting on the explain plan. To get the explain plan you can use this method:
https://github.com/apache/phoenix/blob/master/phoenix-core/src/test/java/org/apache/phoenix/util/TestUtil.java#L1255

Then you can check the explain plan. For server it will have the string "DELETE ROWS SERVER SELECT"